lib/Doctrine/DBAL/Statement.php

```diff
@@ -138,6 +138,8 @@ public function bindValue($name, $value, $type = null)
      */
     public function bindParam($name, &$var, $type = PDO::PARAM_STR, $length = null)
     {
+        $this->params[$name] = $var;
+        $this->types[$name] = $type;
```

Comment:
@deeky666 interestingly, this could probably also have solved the oci8 issues...